rust/lance-linalg/src/kmeans.rs

```diff
@@ -538,12 +539,114 @@ impl KMeans {
     }
 }
 
+/// Return a slice of `data[x,y..y+strip]`.
+#[inline]
+fn get_slice(data: &[f32], x: usize, y: usize, dim: usize, strip: usize) -> &[f32] {
+    &data[x * dim + y..x * dim + y + strip]
+}
+
+/// Fast partition computation for L2 distance.
+fn compute_partitions_l2(centroids: &[f32], data: &[f32], dim: usize) -> Vec<u32> {
+    const STRIPE_SIZE: usize = 128;
+    const TILE_SIZE: usize = 16;
```

Comment:
nit: we may want to benchmark these and use different numbers on different CPUs,

Reply:
one idea is to use something like L1 cache size * factor

Reply:
This is single CPU, so the cache is target to the 32KB / 64KB L1 per core.

Expect to have higher level (i.e., kmeans ) to handle distribution of multiple batches to centroids computation.